### PR TITLE
Editorial: String.prototype.matchAll iterators won't return null

### PR DESCRIPTION
**Pull request by susisu**
Closes https://github.com/tc39/ecma262/issues/3421

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -34894,7 +34894,7 @@ <h1>String.prototype.match ( _regexp_ )</h1>
 
       <emu-clause id="sec-string.prototype.matchall">
         <h1>String.prototype.matchAll ( _regexp_ )</h1>
-        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator. Each iteration result's value is an Array containing the results of the match, or *null* if the String did not match.</p>
+        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator. Each iteration result's value is an Array containing the results of the match.</p>
```
Comment (michaelficarra) [inline spec.html:34897]: This is fine, but I would probably say a bit more.

```suggestion
        <p>This method performs a regular expression match of the String representing the *this* value against _regexp_ and returns an iterator that yields match results. Each match result is an Array containing the matched portion of the String as the first element, followed by the portions matched by any capturing groups. If the regular expression never matches, the returned iterator does not yield any match results.</p>
```

Comment (michaelficarra) [reply spec.html:34897]: We could also `<dfn>` a term for these match result arrays constructed and returned by `RegExpBuiltinExec`. /cc @tc39/ecma262-editors

Comment (michaelficarra) [reply spec.html:34897]: In the editor call today, @syg agreed that we should probably `<dfn>` this thing. We could do it as part of this PR or as a follow-up, doesn't matter to me.

Comment (susisu) [reply spec.html:34897]: Thank you for the review!

I'm totally not sure how to add a definition for the term, so I'd prefer not to address that in this PR.

**Review by michaelficarra — APPROVED** [review]

**Review by bakkot — APPROVED** [review]

**Review by syg — APPROVED** [review]

**Comment by ljharb** [comment]
@susisu would you please [register as a contributor](https://tc39.es/agreements/contributor/)?

**Comment by susisu** [comment]
@ljharb Of course! I submitted the form.

**Pull request merged**